### DISCUSSION TITLE
ollama version 0 ?

### DISCUSSION THREAD

**Discussion by iplayfast** [open-webui/open-webui#540] [General]
# Bug Report

## Description

**Bug Summary:**
Your "effortless setup" is false advertising. 
Basically other then getting a web interface up, I'm finding that it is totally unusable. It does not find my locally installed ollama or find the locally installed models.  I've spent an hour trying various things to get it talking to my working ollama. 

When I run 
./run-ollama-docker.sh 
Do you want ollama in Docker with GPU support? (y/n): y
Error response from daemon: No such container: ollama
latest: Pulling from ollama/ollama
Digest: sha256:74b2ac9790e07ff5871398a75eee42b758c7353ecc6579a4108a4b0de9bd78b2
Status: Image is up to date for ollama/ollama:latest
docker.io/ollama/ollama:latest

What's Next?
  View a summary of image vulnerabilities and recommendations → docker scout quickview ollama/ollama:latest
unknown shorthand flag: ' ' in - -v ollama:/root/.ollama -p 11434:11434 --name ollama ollama/ollama --gpus=all
See 'docker run --help'.
Total reclaimed space: 0B

No idea what this means. I normally run ollama as a server on my system. 
It is running, as I can do "ollama run phi" and be talking to ollama.
However when I go to the web browser at http://localhost:3000/ it finds ollama version 0, says it can't connect and basically is non-responsive. 

Assuming that it is a connection issue I went to the troubleshooting guide and found 
**If you're experiencing connection issues, it’s often due to the WebUI docker container not being able to reach the Ollama server at 127.0.0.1:11434 (host.docker
.internal:11434) inside the container . Use the `--network=host` flag in your docker command to resolve this. Note that the port changes from 3000 to 8080, resulting in the link: `http://localhost:8080`.**

so does that mean I need to edit my run-ollama-docker.sh file? 

I've also tried the run.sh and after downloading a bunch of software when to the system and logging in, found that it still had no connection. 

**Steps to Reproduce:**
git clone 

**Expected Behavior:**
[Describe what you expected to happen.]

**Actual Behavior:**
[Describe what actually happened.]

## Environment

- **Operating System:** [e.g., Windows 10, macOS Big Sur, Ubuntu 20.04]
- **Browser (if applicable):** [e.g., Chrome 100.0, Firefox 98.0]

## Reproduction Details

**Confirmation:**

- [ ] I have read and followed all the instructions provided in the README.md.
- [ ] I have reviewed the troubleshooting.md document.
- [ ] I have included the browser console logs.
- [ ] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]

**Docker Container Logs:**
[Include relevant Docker container logs, if applicable]

**Screenshots (if applicable):**
[Attach any relevant screenshots to help illustrate the issue]

## Installation Method
git clone https://github.com/ollama-webui/ollama-webui.git
chmod +x *.sh
./run-ollama-docker.sh
./run.sh
./run-compose.sh
(all fail)

## Additional Information

[Include any additional details that may help in understanding and reproducing the issue. This could include specific configurations, error messages, or anything else relevant to the bug.]

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by justinh-rahb** [comment]
You can try running the following command instead of using the `run-ollama-docker.sh` script:
```bash
docker run -d -p 3000:8080 -e OLLAMA_API_BASE_URL=http://host.docker.internal:11434/api -v ollama-webui:/app/backend/data --name ollama-webui --restart always ghcr.io/ollama-webui/ollama-webui:main
```
This command will run the Docker container with the necessary configuration to connect to your locally installed Ollama server. Make sure to clean up any existing containers, stacks, and volumes before running this command.

Regarding the troubleshooting guide's recommendation to use the `--network=host` flag, this is only necessary if the WebUI docker container cannot reach the Ollama server at `127.0.0.1:11434 (host.docker.internal:11434)` inside the container. However, since you are running Ollama on your system and not in a Docker container, this is not necessary.

I hope this helps you get Ollama up and running! If you continue to experience issues, please let me know.

**Comment by 0xb15d3** [comment]
You should at least include what command you used to install webui. Is there a reason why you're not using the command provided in the readme?

`./run-ollama-docker.sh` installs ollama on your system not the webui.
